docs/DeveloperGuide.md

```diff
@@ -1,38 +1,321 @@
+
+
+`DeleteDishCommand` is implemented in such a way because:
+1. It promotes loose coupling between components. For instance, `Main` doesn't need to know the details of how the `execute()` of `DeleteDishCommand` is executed or how the message is displayed in `Ui`.
+2. Each component has a specific role and responsibility. `Main` is responsible for receiving user input and invoking `execute()`, `DeleteDishCommand` is responsible for encapsulating the delete operation, `Menu` is responsible for managing the menu items, and `Ui` is responsible for displaying messages to the user. This separation of concerns makes the code more maintainable and easier to understand.
+
+### Edit Price
+
+![Edit Price Execution](images/sequence/EditPriceCommand_execute.png)
+
+*Figure 15: Execution of edit_price command*
+
+API: [EditPriceCommand.java]({repoURL}src/main/java/seedu/cafectrl/command/EditPriceCommand.java)
+
+When the `execute()` method of `EditPriceCommand` is invoked in `Main`, it subsequently calls the `setPrice()` method on the `Dish` object to modify the price of the specific dish. Following this, the `showEditPriceMessages()` method in the `Ui` component is triggered to retrieve and display a message from `Messages` related to the successful execution of the price modification process. This sequence of actions orchestrates the flow of information and operations between the `Main`, `EditPriceCommand`, `Dish`, and `Ui` components, ensuring the seamless handling of the price editing functionality within the application.
+
+### Help
+
+![Help Execution](images/sequence/HelpCommand_execute.png)
+
+*Figure 16: Execution of help command*
+
+API: [HelpCommand.java]({repoURL}src/main/java/seedu/cafectrl/command/HelpCommand.java)
+
+When the `execute()` method of `HelpCommand` is invoked in `Main`, it subsequently calls the `showHelp()` method in `Ui`. In `showHelp()`, messages related to command usage will be retrieved and be printed out using by self-invoking `showToUserWithSpaceInBetweenLines(messages: String...)`.
+
+--------------------------------------------------------------------------------------------------------------------
+## **Product scope**
+### Target user profile
+
+Café proprietors who ***love*** typing on CLI and are seeking for a software solution to optimize the management of their café's operations.
+
+### Value proposition
 
-## Glossary
+Our product aims to optimize managing of inventory and cash flow in a restaurant. Our CLI platform empowers users to streamline stock inventory, menu and orders. Users will also briefly be able to gain valuable insights through comprehensive sales reporting, enabling them to analyze sales trends and calculate revenue/profit margins, eliminating the need for cross-platform management.
 
-* *glossary item* - Definition
+### User stories
```

Comment:
The user stories are clear and considerate enough to make sure all kinds of user can realize their needs.

Reply:
Thank you!